src/mpyl/utilities/docker/__init__.py

```diff
@@ -338,3 +340,21 @@ def remove_container(logger: Logger, container: Container) -> None:
     logger.debug(f"Removing container {container.id}")
     docker.remove(container.id)
     logger.info(f"Removed container {container.id}")
+
+
+def check_ecr_repo(logger: Logger, repo):
+    # Initialize the ECR client
+    ecr_client = boto3.client('ecr')
+
+    # Check if the repository already exists
+    try:
```

Comment:
Instead of a try catch structure, don't pass any params to the `describe_repositories()` call. According to the documentation that will list all repo's, so then you can search through the list and handle accordingly. Much cleaner piece of code that way!

Reply:
I can do that as well, but doesn't `ecr_client.describe_repositories(repositoryNames=[repo,])` already does this? I don't see how it is cleaner.

Reply:
I think we can keep the try-catch, otherwise we need to implement pagination as well (according to their documentation it returns 100 repos at once normally). You can set the max value to a 1000 but then the call would take longer.. Having a small try-catch block with a specific error type should be fine in my opinion.